kubectl-cloudflow/cmd/deploy_command.go

```diff
@@ -373,3 +376,42 @@ func validateDeployCmdArgs(cmd *cobra.Command, args []string) error {
 
 	return nil
 }
+
+// The function validates that the operators for Spark and Flink is installed if the application uses any of those streamlet types
+func validateStreamletRunnersDependencies(applicatonSpec cfapp.CloudflowApplicationSpec) {
+
+	const requiredSparkVersion = "v1beta2"
+	const requiredFlinkVersion = "v1beta1"
+
+	runnerType := func(runnerTypeName string) bool {
+		for _, v := range applicatonSpec.Streamlets {
+			if v.Descriptor.Runtime == runnerTypeName {
+				return true
+			}
+		}
+		return false
+	}
+
+	validateRunnerType := func(crdName string, prettyName string, expectedVersion string) error {
+		version, err := exec.Command("kubectl", "get", "crds", crdName, "-o", "jsonpath={.spec.version}").Output()
+		if err != nil {
+			return fmt.Errorf("cannot detect that %s is installed, please install %s before continuing (%v)", prettyName, prettyName, err.Error())
+		}
+		if string(version) != expectedVersion {
+			return fmt.Errorf("%s is installed but the wrong version, required %s, installed %s", prettyName, requiredSparkVersion, string(version))
+		}
+		return nil
+	}
+
+	if runnerType("spark") {
```

Comment:
Here is another idea to do this: What if you would find the distinct runner names in the streamlets and then for every runtime found just do validateRunnerType, passing in the runtime found? (and have a const map, runtime -> crd, so you just get that crd from the map by runtime key)

Reply:
ok, its an alternative